enumerables.rb

```diff
@@ -1,37 +1,72 @@
 module Enumerables
+
   def my_each
-    count = 0
-    while count < self.size do
-      yield(self[count])
-      count += 1
+    return to_enum unless block_given?
+
+    for i in 0..self.length
+      yield(self[i])
     end
-    self
   end
 
   def my_each_with_index
-    
+    return to_enum unless block_given?
+    my_index = 0
+    self.my_each do |i|
+      yield i, my_index
+      my_index += 1
+    end
   end
 
   def my_select
     my_arr = []
-    count = 0
-    while count < self .length
-      my_arr << (self[count]) if yield(self[count]) == true
-      count += 1
-    end
+    self.my_each{ |x| my_arr << x if yield(x) }
     my_arr
   end
 
-  def my_all?
-  end
+    def my_all?
+      self.my_each do |index|
+        return true unless block_given?
+        true_false = yield(index)
+        return false unless true_false
+      end
+      true
+      to_enum
+    end
```

Comment:
- [ ]  enumerables #my_all? returns true if the block never returns false or nil
      Failure/Error: let(:true_block) { proc { |num| num <= HIGHEST_VALUE } }

- [ ] enumerables #my_all? returns false if the block returns false or nil::range
      Failure/Error: expect(range.my_all?(&false_block)).to eq(range.all?(&false_block))

- [ ] enumerables #my_all? does not mutate the original array
      Failure/Error: array.my_all? { |num| num + 1 }

- [ ] enumerables #my_all? when no block or argument is given returns false when one of the collection members are false or nil
      Failure/Error: expect(false_array.my_all?).to be false_array.all?
      
        expected false
             got true

- [ ]  enumerables #my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Integer
      Failure/Error: expect(array.my_all?(Integer)).to be array.all?(Integer)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)

- [ ]  enumerables #my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Numeric
      Failure/Error: expect(numbers.my_all?(Numeric)).to be numbers.all?(Numeric)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)

- [ ]  enumerables #my_all? when a class is passed as an argument returns false if any of the collection is not a member of such class
      Failure/Error: expect(array.my_all?(Integer)).to be array.all?(Integer)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)

- [ ] enumerables #my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
      Failure/Error: expect(words.my_all?(/d/)).to be words.all?(/d/)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)

- [ ] enumerables #my_all? when a Regex is passed as an argument returns false if any of the collection does not match the Regex
      Failure/Error: expect(words.my_all?(/o/)).to be words.all?(/o/)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)

- [ ]  enumerables #my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
      Failure/Error: expect(array.my_all?(3)).to be array.all?(3)

- [ ] enumerables #my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern
      Failure/Error: expect(array.my_all?(3)).to be array.all?(3)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)